screens/network/join_host.gui_script

```diff
@@ -0,0 +1,82 @@
+local monarch = require "monarch.monarch"
+local url = require "utils.url"
+local gooey = require "gooey.gooey"
+local gooey_buttons = require "utils.gooey_buttons"
+local button = require "utils.button"
+local netcore = require "pokedex.network.netcore"
+local gooey_input_text = require "utils.gooey_input_text"
+local gui_colors = require "utils.gui_colors"
+local messages = require "utils.messages"
+
+local function redraw_buttons(self)
+	self.has_selected_ip = self.ip ~= nil and self.ip ~= ""
+	
+	local color_txt = gui_colors.BUTTON_TEXT_DISABLED
+	local flipbook = "common_disabled"
+	if self.has_selected_ip then
+		color_txt = gui_colors.BUTTON_TEXT
+		flipbook = "common_up"
+	end
+	
+	gui.set_color(gui.get_node("txt_join_host"), color_txt)
+	gui.play_flipbook(gui.get_node("btn_join_host"), flipbook)
+end
+
+local function apply_port_text(self, text)
+	local as_number = tonumber(text)
+	if as_number then
+		self.port = as_number
+	end
+end
+
+local function apply_ip_text(self, text)
+	self.ip = text
+	redraw_buttons(self)
+end
+
+local function redraw(self)
+	redraw_buttons(self)
+end
+
+function init(self)
+	msg.post(url.MENU, messages.HIDE)
+	button.acquire()
+
+	self.has_selected_ip = false
+
+	self.ip = netcore.get_default_connect_address() or ""
+	self.port = netcore.get_default_connect_port()
+	self.ip_input = gooey_input_text.create("txt_ip", "ip_cursor", tostring(self.ip), function(text) return apply_ip_text(self, text) end, {max_length = 20, allowed_characters="[%a%d.-]"})
+	self.port_input = gooey_input_text.create("txt_port", "port_cursor", tostring(self.port), function(text) return apply_port_text(self, text) end, {max_length = 6, allowed_characters="[%d]"})
+
+	redraw(self)
+end
+
+function final(self)
+end
+
+function on_message(self, message_id, message)
+end
```

Comment:
Can remove these

Reply:
I actually would rather have them in than not in, in general. In my mind they're part of the public API of a gui_script so they should be there despite currently remaining unused.